gridstatus/ieso.py

```diff
@@ -0,0 +1,473 @@
+            values="EnergyMW",
+            aggfunc="first",
+        ).reset_index()
+
+        pivot_df["Published Time"] = pd.Timestamp(
+            published_time,
+            tz=self.default_timezone,
+        )
+
+        pivot_df = utils.move_cols_to_front(
+            pivot_df,
+            [
+                "Interval Start",
+                "Interval End",
+                "Published Time",
+                "Ontario",
+            ],
+        )
+
+        pivot_df.columns.name = None
+
+        col_mapper = {
+            col: f"{col} Load Forecast" for col in ["Ontario", "East", "West"]
+        }
+
+        pivot_df = pivot_df.rename(columns=col_mapper)
+
+        # If no end is provided, return data from single date
+        if not end:
+            return pivot_df[
+                pivot_df["Interval Start"].dt.date == date_only
+            ].reset_index(drop=True)
+
+        end_date_only = pd.to_datetime(end).date()
+
+        return pivot_df[
+            (pivot_df["Interval Start"].dt.date >= date_only)
+            & (pivot_df["Interval End"].dt.date <= end_date_only)
+        ].reset_index(drop=True)
+
+    # TODO add fuel mix. http://reports.ieso.ca/public/GenOutputbyFuelHourly/
+    def get_fuel_mix(self, date, end=None, verbose=False):
+        pass
+
+    def _today(self):
+        return pd.Timestamp.now(tz=self.default_timezone).date()
+
+    # Function to extract data for a specific Market Quantity considering namespace
+    def _extract_load_in_market_quantity(
+        self,
+        market_quantity_element,
+        market_quantity_name,
+    ):
+        for mq in market_quantity_element.findall("MQ", NAMESPACES_FOR_XML):
+            market_quantity = mq.find("MarketQuantity", NAMESPACES_FOR_XML).text
+
+            if market_quantity_name in market_quantity:
+                return mq.find("EnergyMW", NAMESPACES_FOR_XML).text
+
+        return None
+
+    # Function to find all triples of 'Interval', 'Market Total Load', and
+    # 'Ontario Load' in the XML file
+    def _find_loads_at_each_interval_from_xml(self, root_element):
+        interval_load_demand_triples = []
+
+        for interval_energy in root_element.findall(
+            "DocBody/Energies/IntervalEnergy",
+            NAMESPACES_FOR_XML,
+        ):
+            interval = interval_energy.find("Interval", NAMESPACES_FOR_XML).text
+            market_total_load = self._extract_load_in_market_quantity(
+                interval_energy,
+                "Total Energy",
+            )
+            ontario_load = self._extract_load_in_market_quantity(
+                interval_energy,
+                "ONTARIO DEMAND",
+            )
+
+            if market_total_load and ontario_load:
+                interval_load_demand_triples.append(
+                    [int(interval), float(market_total_load), float(ontario_load)],
+                )
+
+        return interval_load_demand_triples
+
+    def _handle_input_date(self, date):
```

Comment:
try to use `utils._handle_date` rather than define this helper

Reply:
Got it. `utils._handle_date` doesn't work for the hourly data though. If you pass in "today" you'll get out a timestamp at the beginning of the day and then you'll end up only loading the first hour of the day's data.

Reply:
am i seeing correctly this is only called in one place (get_load_forecast) and in that place, this hourly issue doesnt seem to apply?

Reply:
I've removed this method in favor of `utils._handle_date` (which I think shouldn't have a leading underscore if it's used outside the utils file).

Reply:
ya, you're right about that underscore. even worse i think i use it in the close source code base too, which is why ive been hesitant to change it. afraid of breaking things